Updating to work with latest Go 1.19 version

donpui opened the pull request:
Updating to work with latest Go 1.19 version.

Copy wasm_exec.js was taken from GOROOT/misc/wasm folder of version 1.19.2

Ticket #OP2300

donpui commented:
@JustusFT this will require your look, as we don't pass E2E. Running locally also seems we fail with worker, when we try to use it.

meejah commented:
Having our own vendored copy of `wasm_exec.js` is bad; can we not import it from Go "directly" or something?

donpui commented:
> Having our own vendored copy of `wasm_exec.js` is bad; can we not import it from Go "directly" or something?

My initial thought the same, however I am not sure if it is possible as application and worker runs on client side where is no go environment. Seems this file has to be copied https://github.com/golang/go/wiki/WebAssembly#webassembly-in-chrome. This part requires more investigation.

meejah commented:
> Seems this file has to be copied [..]

Having a copy is the artifacts is fine, but can't we just grab it from the "go root" at build time? (i.e. instead of burning it into this repository?)

vu3rdd commented:
@donpui May be once https://github.com/LeastAuthority/wormhole-william/pull/92 is merged in, you may update the submodule to the latest commit in this PR itself, so all the Go 1.19 changes would be visible from this PR.

donpui commented:
I have updated with w-w changes and also moved wasm_exec.js. Now it will be copied during the build with small change in file. @JustusFT take a look, maybe it can be done even better.

After all changes, seems we have working solution with Go-1.19

JustusFT commented on client/gulpfile.js:
```diff
@@ -85,13 +85,26 @@ const javascriptWatch = () =>
+    execSync(
+      "cp \"$(go env GOROOT)/misc/wasm/wasm_exec.js\" src/worker"
+    );
+    // export as Go module
```
LGTM, however I'm not sure we need to modify the file with an export anymore. Maybe we can change from `import Go from "wasm_exec";` to `import "wasm_exec";` since loading the file already sets Go as a global variable. Though we would need some changes to the test suite since some tests import go as well. But this PR is good already. We could update the import and tests in a separate PR :+1:

donpui replied:
I have updated and all tests pass, so merging.
